docs/source/algorithms/SaveP2D-v1.rst

```diff
@@ -0,0 +1,107 @@
+multidimensional Rietveld refinements. 
+The input for this algorithm needs to be a 2D workspace containing information about dSpacing and 
+dSpacingPerpendicular. A 2D workspace can be created using the :ref:`Bin2DPowderDiffraction
+<algm-Bin2DPowderDiffraction>` algorithm. 
+The input values ``removeNaN`` and ``removeNegatives`` control whether intensity values that are negative 
+or NaN, respectively, are automatically removed from the dataset. 
+``RemoveNegatives`` also removes intensities equal to zero.
+Turning ``cutDdata`` on, allows to cut the measuring data to the specified ranges of 
+theta, lambda, dSpacing and dSpacingPerpendicular. 
+
+Output
+######
+
+The output file contains a short comment header giving the title, the instrument parameter file, 
+the binning parameters of the workspace and the used instrument/detector bank.
+Thereafter the measuring data is written into 5 columns, namely, 2theta, lambda, dSpacing,
+dSpacingPerpendicular and intensity.
+
+Usage
+-----
+
+**Example: Create a ".p2d" file from a 2D Workspace. Remember to change the Filepath for the OutputFile!**
+
+.. testcode:: SaveP2D
+
+	# create a 2D Workspace
+    # repeat this block for each spectrum
+    xData = [1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0]		     # d values for one spectrum (one dPerpendicular value)
+    yData = ['1','2','3','4']					             # dPerpendicular binedges
+    zData = [1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0]		     # intensity values
+    eData = [1,1,1,1,1,1,1,1,1]                              # error values
+
+    # used to join all spectra
+    xDataTotal = []					                         # d Values for all spectra
+    zDataTotal = []					                         # intensity values for all spectra
+    eDataTotal = []                                          # error values for all spectra
+    nSpec = len(yData)-1                                     # number of spectra
+    
+    # Create d and intensity lists for workspace
+    for i in range(0,nSpec):
+        xDataTotal.extend(xData)	   # extends the list of x values in accordance to the number of spectra used
+        zDataTotal.extend(zData)	   # extends the list of intensity values in accordance to the number of spectra used
+        eDataTotal.extend(eData)       # extends the list of error values in accordance to the number of spectra used
+    
+    # Create a 2D Workspace containing d and dPerpendicular values with intensities
+    CreateWorkspace(OutputWorkspace = 'Usage_Example', DataX = xDataTotal, DataY = zDataTotal, DataE = eDataTotal, WorkspaceTitle = 'test', NSpec = nSpec, UnitX = 'dSpacing', VerticalAxisUnit = 'dSpacingPerpendicular', VerticalAxisValues = yData)
+    # Create a .p2d file containing the testdata	
+    SaveP2D(Workspace = "Usage_Example", OutputFile = "Usage_Example", RemoveNaN = False, RemoveNegatives = False, CutData = False)
+	
+Output:
+
+.. testoutput:: SaveP2D
```

Comment:
So this output is actually the output from the python console, so for your algorithm its something like 
Exporting:
```
C:\........\filename`

   0%
  33%
  67%


Exported: C:\.....\filename
```
We should structure this like SaveAscii https://raw.githubusercontent.com/mantidproject/mantid/master/docs/source/algorithms/SaveAscii-v2.rst so the generated file is cleaned up using the cleanup method.

In order to test the output we'll need to add an option to the test output using

```suggestion
.. testoutput:: SaveP2D
   :options: +ELLIPSIS
```

And finally the testouput code should read
```
Exporting: ...

   0%
  33%
  67%


Exported: ...
```
Where we have used the ellipsis to signify any output, as this location will be different on each os.

Reply:
I already changed some stuff concerning this problem and tried to create a similiar output to the one I expect by printing the file line by line. I additionally added your recommendations.